File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        if ch not in word:
            return word
        res=""
        st=[]
        ind=word.index(ch)
        for i in range(ind+1):
            st.append(word[i])
        while st:
            res+=st.pop()
        for i in range(ind+1,len(word)):
            res+=word[i]
        return res

File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        String res="";
        int ind=word.indexOf(ch);
        if(ind==-1){
            return word;
        }
        Stack<Character> st=new Stack<>();
        for(int i=0;i<=ind;i++){
            st.push(word.charAt(i));
        }
        while(!st.isEmpty()){
            res+=st.pop();
        }
        for(int j=ind+1;j<word.length();j++){
            res+=word.charAt(j);
        }
        return res;
    }
}